### ISSUE TITLE
Unable to use keyboard to interact with editor gutter controls

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#13787]
### System details

    RStudio Edition : Desktop or Server
    RStudio Version : 2023.12 (must include changes from #13786)
    OS Version      : 
    R Version       : 

### Steps to reproduce the problem

- Start screen reader
- With a build containing changes from #13786, turn on screen reader support (Help / Accessibility / Screen Reader Support).
- Create a source file that has some collapsable regions, for example File / New File / C++ file.
- Ctrl+1 to focus the editor
- Press ESC to escape the editor (you'll see a focus rectangle around it and hear an announcement)
6. Shift+Tab to move to the gutter (you'll see a focus rectangle around it and hear an announcement)
7. As instructed press Enter to try and interact with the gutter

### Describe the problem in detail

Nothing happens.

### Describe the behavior you expected

Focus should go to one of the gutter controls (e.g. expand/collapse widget) and up/down arrows should allow moving between them and enter or spacebar should toggle them.

You can see this working on the Ace kitchen sink: https://ace.c9.io/build/kitchen-sink.html. Be sure "Keyboard Accessibility Mode" is checked.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by gtritchie** [comment]
I won't have time to look into this for this release. It's not urgent but should definitely be done, so moving to next release.
